crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,503 @@
+    ///
+    /// Given a target ending pool share reserves, $z_t$, and bond reserves, $y_t$,
+    /// the trade deltas to achieve that state would be:
+    ///
+    /// $$
+    /// \Delta x = c * (z_t - z_e) \\
+    /// \Delta y = y - y_t - c(\Delta x)
+    /// $$
+    ///
+    /// where $c$ is the vault share price and
+    /// $c(\Delta x)$ is the (open_long_curve_fee)[long::fees::open_long_curve_fees].
+    fn trade_deltas_from_reserves(
+        &self,
+        share_reserves: FixedPoint,
+        bond_reserves: FixedPoint,
+    ) -> (FixedPoint, FixedPoint) {
+        let base_delta =
+            (share_reserves - self.effective_share_reserves()) * self.vault_share_price();
+        let bond_delta =
+            (self.bond_reserves() - bond_reserves) - self.open_long_curve_fees(base_delta);
+        (base_delta, bond_delta)
+    }
+
+    /// Calculates the pool reserve levels to achieve a target interest rate.
+    /// This calculation does not take Hyperdrive's solvency constraints or exposure
+    /// into account and shouldn't be used directly.
+    ///
+    /// The price for a given fixed-rate is given by $p = 1 / (r t + 1)$, where
+    /// $r$ is the fixed-rate and $t$ is the annualized position duration. The
+    /// price for a given pool reserves is given by $p = \frac{\mu z}{y}^T$,
+    /// where $\mu$ is the initial share price and $T$ is the time stretch
+    /// constant. By setting these equal we can solve for the pool reserve levels
+    /// as a function of a target rate.
+    ///
+    /// For some target rate, $r_t$, the pool share reserves, $z_t$, must be:
+    ///
+    /// $$
+    /// z_t = \frac{1}{\mu} \left(
+    ///   \frac{k}{\frac{c}{\mu} + \left(
+    ///     (r_t t + 1)^{\frac{1}{T}}
+    ///   \right)^{1 - T}}
+    /// \right)^{\tfrac{1}{1 - T}}
+    /// $$
+    ///
+    /// and the pool bond reserves, $y_t$, must be:
+    ///
+    /// $$
+    /// y_t = \left(
+    ///   \frac{k}{ \frac{c}{\mu} +  \left(
+    ///     \left( r_t t + 1 \right)^{\frac{1}{T}}
+    ///   \right)^{1-T}}
+    /// \right)^{1-T} \left( r_t t + 1 \right)^{\frac{1}{T}}
+    /// $$
+    fn reserves_given_rate_ignoring_exposure<F: Into<FixedPoint>>(
+        &self,
+        target_rate: F,
+    ) -> (FixedPoint, FixedPoint) {
+        let target_rate = target_rate.into();
+
+        // First get the target share reserves
+        let c_over_mu = self
+            .vault_share_price()
+            .div_up(self.initial_vault_share_price());
+        let scaled_rate = (target_rate.mul_up(self.annualized_position_duration()) + fixed!(1e18))
+            .pow(fixed!(1e18) / self.time_stretch());
+        let target_base_reserves = (self.k_down()
+            / (c_over_mu + scaled_rate.pow(fixed!(1e18) - self.time_stretch())))
+        .pow(fixed!(1e18) / (fixed!(1e18) - self.time_stretch()));
+        let target_share_reserves = target_base_reserves / self.initial_vault_share_price();
```

Comment:
This is confusing to me since we can't convert between base and shares using the initial share price. I see why we do this (and I am glad that we don't recompute the bond reserves from scratch!). This is a case in which I'd advocate for calling the `target_base_reserves` something like `inner`. I see the appeal of this naming, but it isn't accurate IMO.

Reply:
good catch!